agent/agentbootstrap/bootstrap.go

```diff
@@ -108,11 +111,23 @@ func InitializeState(
 	info.Tag = nil
 	info.Password = c.OldPassword()
 
+	// Add the controller model cloud and credential to the database.
+	cloudCred, cloudCredTag, err := getCloudCredential(adminUser, args)
```

Comment:
This is existing code which has been tweaked to fix the current work. There's only so many yaks one can shave in a single PR. The PR is large enough as it is.

Reply:
Actually, the way it behaves is correct according to current usage - it returns 2 values, the credential tag composed from the passed in args, plus, if available, the credential. It's possible that the credential tag is correctly formed, but there's no matching credential. So in that case it's not really valid to return an error because the credential tag result is still valid. The method itself should be split up to deal with the 2 concerns separately. The original implementation didn't make sense anyway as it always returned a map with a single key value. So this chunk of logic needs a rewrite.